src/harness/CMakeLists.txt

```
@@ -61,14 +61,26 @@ target_sources(harness PRIVATE
 )
 
 if (IO_PLATFORM STREQUAL "SDL_OpenGL")
+    # generate embedded glsl resources
+    foreach(elem resources/framebuffer_vert.glsl resources/framebuffer_frag.glsl resources/3d_vert.glsl resources/3d_frag.glsl)
+        add_custom_command(
+            OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${elem}.h"
+            COMMAND ${CMAKE_COMMAND} -DSOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR} -DFILE="${elem}" -P ${CMAKE_SOURCE_DIR}/cmake/EmbedResource.cmake
```

`CMAKE_SOURCE_DIR` ponit to the root of a build tree, where we want the root of the the dethrace project.

```suggestion
            COMMAND "${CMAKE_COMMAND}" "-DSOURCE_DIR=${dethrace_SOURCE_DIR}" "-DFILE=${elem}" -P "${dethrace_SOURCE_DIR}/cmake/EmbedResource.cmake"
```

---

I understand the `"`, but why `dethrace_SOURCE_DIR`? These are resources specific to `harness`, and should only be available to `#include` from harness code?

---

Looking at it, use of `CMAKE_CURRENT_SOURCE_DIR` is correct.
But `CMAKE_SOURCE_DIR` isn't: it should be either `PROJECT_SOURCE_DIR` or `dethrace_SOURCE_DIR`.

---

are you suggesting to move `cmake/EmbedResource.cmake` to `src/DETHRACE/cmake/EmbedResource.cmake`?

---

No, if you happen to include dethrace as a subproject for your own project, `CMAKE_SOURCE_DIR` points to the root source directory. Not to the root directory of dethrace.

---

ah, yep, I got it I think.